### PR TITLE
Removing skip attribute from color validation tests

### LINKED ISSUE #11948: Fix the rest of the Color validating tests
### Description

A lot of the color checking tests weren't correctly awaiting the validation calls, so they weren't really validating anything. The test would just exit before the Assert would happen.  

I've fixed a number of them in this PR [here](https://github.com/dotnet/maui/pull/11016) and then skipped some because that PR was becoming too big. 

I've marked all skipped tests with 

```C#
[Fact(DisplayName = "<TEST NAME>", 
    Skip = "This test is currently invalid https://github.com/dotnet/maui/issues/11948")]
```

You can use that to locate the skipped tests that need to be reworked. I tested a few of these in real apps and, so far, didn't find any bugs. It just seems like the tests aren't quite written correctly.

### PR DESCRIPTION
**Pull request by MSLukeWest**
### Description of Change

Re-enabling and fixing some of the color validation tests.  Some of them had to be left disabled for various reasons, see follow-up issues below.

### Issues Fixed

Some color validation tests are currently disabled.

Partially Fixes #11948 

### Follow-up Issues Opened

#13507 
#13693 
#13692

### CODE REVIEW

File: src/Core/src/Platform/iOS/SearchBarExtensions.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal static void UpdateBackground(this UISearchBar uiSearchBar, ISearchBar s
 			var background = searchBar.Background;
 
 			if (background is SolidPaint solidPaint)
-				uiSearchBar.BarTintColor = solidPaint.Color.ToPlatform();
+				ViewExtensions.UpdateBackground(uiSearchBar, solidPaint);
```
Comment (PureWeen) [inline src/Core/src/Platform/iOS/SearchBarExtensions.cs:24]: I'm thinking we might want to split this change out from this PR in order to fix how the background is applied.

If you compare android to ios the SearchBar on iOS is more of a tint
image

```XAML
 <VerticalStackLayout>
        <SearchBar Background="Purple"></SearchBar>
        <SearchBar>
            <SearchBar.Background>
                <LinearGradientBrush EndPoint="1,0">
                    <GradientStop Color="Yellow" 
                                      Offset="0.1" />
                    <GradientStop Color="Green"                                   
                                      Offset="1.0" />
                </LinearGradientBrush>
            </SearchBar.Background>
        </SearchBar>
        <SearchBar></SearchBar>
        
    </VerticalStackLayout>
</ContentPage>
```

I think to get the searchbar background to match you have to set the background color on one of the inner controls
https://stackoverflow.com/a/48204512

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/src/Platform/iOS/SearchBarExtensions.cs (hunk 1 above)
Comment (MSLukeWest) [reply src/Core/src/Platform/iOS/SearchBarExtensions.cs]: Ok, thanks for the comment.  For now I'm undoing that change and leaving the test disabled, with a link to new issue https://github.com/dotnet/maui/issues/13693

File: src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBaseOfT.cs [hunk 2]
```diff
@@ -178,9 +178,9 @@ protected Task SetValueAsync<TValue>(IView view, TValue value, Action<THandler,
 			Assert.Equal(initialNativeVal, newNativeVal);
 		}
 
-		protected Task ValidateHasColor(IView view, Color color, Action action = null) =>
-			ValidateHasColor(view, color, typeof(THandler), action);
-			
+		protected Task ValidateHasColor(IView view, Color color, Action action = null, string updatePropertyValue = null) =>
```
Comment (PureWeen) [inline src/Core/tests/DeviceTests.Shared/HandlerTests/HandlerTestBaseOfT.cs:181]: <3

File: src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.cs [hunk 3]
```diff
@@ -120,16 +120,21 @@ public async Task MaxLengthInitializesCorrectly(int maxLength)
...
 			var searchBar = new SearchBarStub()
 			{
-				CancelButtonColor = Colors.MediumPurple
+				Text = "Search",
+				Width = 200,
+				CancelButtonColor = Colors.Yellow,
 			};
 
-			await ValidateHasColor(searchBar, Colors.MediumPurple, () => searchBar.CancelButtonColor = Colors.MediumPurple);
+			await ValidateHasColor(searchBar, Colors.Yellow, () => searchBar.CancelButtonColor = Colors.Yellow, nameof(searchBar.CancelButtonColor));
```
Comment (PureWeen) [inline src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.cs:137]: I think for this test we just want to use

```C#
await ValidateHasColor(searchBar, Colors.Yellow);
```

This test is validating that the color is `initialized` correctly not that it's updated.

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.cs (hunk 3 above)
Comment (MSLukeWest) [reply src/Core/tests/DeviceTests/Handlers/SearchBar/SearchBarHandlerTests.cs]: Thanks for noticing, I fixed this and confirmed the test still passes

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
This PR only touches our tests so we should backport!

**Comment by rmarinho** [comment]
Don't forget to pull fix for tests in older iOS version, as we saw it broke main

**Comment by jfversluis** [comment]
@PureWeen you were taking this one?

**Pull request merged**